src/pages/api/checkout_sessions.ts

```diff
@@ -1,27 +1,41 @@
-/* eslint-disable */
-
+import { type NextApiRequest, type NextApiResponse } from "next";
 import Stripe from "stripe";
 
+import { type CartItem } from "~/types/cart";
+
 const stripe = new Stripe(process.env.STRIPE_SECRET_KEY || "", { apiVersion: "2022-11-15" });
 
-export default async function handler(req: any, res: any) {
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
   if (req.method === "POST") {
     try {
+      const cartItems = req.body as CartItem[];
+
       // Create Checkout Sessions from body params.
       const session = await stripe.checkout.sessions.create({
-        line_items: [
-          {
-            // Provide the exact Price ID (for example, pr_1234) of the product you want to sell
-            price: "price_1N3B5jFqEYfFMGYQwuq3qMyi",
-            quantity: 1,
-          },
-        ],
+        line_items: cartItems.map((product: CartItem) => {
+          return {
+            price_data: {
+              currency: "usd",
+              product_data: {
+                name: product.name,
+                description: product.variant.options.reduce(
+                  (acc: string, option: { value: string }) => `${acc} ${option.value} /`,
```

Comment:
minor thing, but you can avoid the trailing `/` here, maybe add the values to an array, then `arr.join(' / ')`